File: api/models/parameter_history.py
from pydantic import BaseModel, Field
from typing import Any, Optional

class HistoryEntryCreate(BaseModel):
    upload_id: str = Field(..., min_length=1)
    measurement_id: Optional[str] = None
    tab: str = Field(..., pattern="^(intensity|lifetime|correlation|grouping|raster|spectra)$")
    parameter: str  = Field(..., min_length=1, max_length=100)
    old_value: Any = None
    new_value: Any = Field(...)

File: frontend/types/parameterHistory.ts
export type HistoryTab = | "intensity" | "lifetime" | "correlation" | "grouping" | "raster" | "spectra";

export interface HistoryEntry {
    id: string;
    workspace_id: string;
    upload_id: string;
    measurement_id: string | null;
    tab: HistoryTab;
    parameter: string;
    old_value: any;
    new_value: any;
    author_id: string;
    created_at: string;
}